router/schema.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-sql"
+	"github.com/flynn/flynn/pkg/postgres"
+)
+
+func migrateDB(db *sql.DB) error {
+	m := postgres.NewMigrations()
+	m.Add(1,
+		`CREATE EXTENSION IF NOT EXISTS "uuid-ossp"`,
+		`CREATE FUNCTION set_updated_at_column() RETURNS TRIGGER AS $$
+	BEGIN
+		NEW.updated_at = CURRENT_TIMESTAMP AT TIME ZONE 'UTC';
+		RETURN NEW;
+	END;
+$$ language 'plpgsql'`,
+
+		// tcp routes
+
+		`
+CREATE TABLE tcp_routes (
+	id uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+	parent_ref varchar(255) NOT NULL,
+	service varchar(255) NOT NULL,
+	port integer NOT NULL,
+	created_at timestamptz NOT NULL DEFAULT now(),
+	updated_at timestamptz NOT NULL DEFAULT now(),
+	deleted_at timestamptz
+)`,
+		`
+CREATE UNIQUE INDEX tcp_routes_port_key ON tcp_routes
+USING btree (port) WHERE deleted_at IS NULL`,
+		`
+CREATE TRIGGER set_updated_at_tcp_routes
+	BEFORE UPDATE ON tcp_routes FOR EACH ROW
+	EXECUTE PROCEDURE set_updated_at_column()`,
+		`
+CREATE OR REPLACE FUNCTION notify_tcp_route_update() RETURNS TRIGGER AS $$
+BEGIN
+	PERFORM pg_notify('tcp_routes', NEW.id::varchar);
+	RETURN NULL;
+END;
+$$ LANGUAGE plpgsql`,
+		`
+CREATE TRIGGER notify_tcp_route_update
+	AFTER INSERT OR UPDATE OR DELETE ON tcp_routes
+	FOR EACH ROW EXECUTE PROCEDURE notify_tcp_route_update()`,
+
+		// http routes
+
+		`
+CREATE TABLE http_routes (
+	id uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+	parent_ref varchar(255) NOT NULL,
+	service varchar(255) NOT NULL CHECK (service <> ''),
+	domain varchar(255) NOT NULL,
```

Comment:
Seems like a few other fields should get the not empty check, (`domain` here, `service`/`port` for TCP)